src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureKeyVaultReference/AzureKeyVaultSecretProvider.cs

```diff
@@ -63,6 +82,38 @@ public async Task<string> GetSecretValue(Uri secretUri, CancellationToken cancel
             return secretValue;
         }
 
+        internal bool AnyExpiredSecrets()
+        {
+            bool shouldRefreshKeyVaultSecrets = false;
+            List<CachedKeyVaultSecret> secretsToBeRemovedFromCache = new List<CachedKeyVaultSecret>();
+
+            foreach (var cachedSecret in _cachedKeyVaultSecrets)
+            {
+                // Skip the refresh for this key vault secret if it has no expiration time or if it hasn't expired yet
+                if (cachedSecret.ExpiresOn == null || DateTimeOffset.UtcNow < cachedSecret.ExpiresOn)
```

Comment:
I might miss something. I thought we didn't make decisions based on expiration anymore.

Reply:
Now I see - it's the cache expiration time, not the secret :)

Reply:
Renamed to `RefreshAt` so that it's less confusing.